compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5471,8 +5471,11 @@ static TR::Register *inlineArrayCmpP10(TR::Node *node, TR::CodeGenerator *cg, bo
    TR::LabelSymbol *resultLabel = generateLabelSymbol(cg);
 
    bool is64bit = cg->comp()->target().is64Bit();
+   bool isLengthNode64bit = lengthNode->getDataType().isInt64();
+   TR_ASSERT_FATAL_WITH_NODE(lengthNode, lengthNode->getDataType().isInt64() || lengthNode->getDataType().isInt32(),
+                             "lengthNode not int64 or int32. Datatype: %s", node->getDataType().toString());
```

Comment:
provides more precise assert ... need to cover the case: no 64bit length node on 32bit platforms.